.bazelrc

```diff
@@ -0,0 +1,68 @@
+# use a compiler name that doesn't symlink to ccache
+build --action_env=CC=clang-16 --host_action_env=CC=clang-16
+build --action_env=CXX=clang++-16 --host_action_env=CXX=clang++-16
+build --action_env=BAZEL_COMPILER=clang
+build --cxxopt=-std=c++20 --host_cxxopt=-std=c++20
+build --linkopt -fuse-ld=lld
+
+# https://github.com/bazelbuild/rules_foreign_cc/issues/1065
+# https://github.com/bazelbuild/rules_foreign_cc/issues/1186#issuecomment-2053550487
+build --host_action_env=CXXFLAGS=-Wno-int-conversion
+build --action_env=CXXFLAGS=-Wno-int-conversion
+build --host_action_env=CFLAGS=-Wno-int-conversion
+build --action_env=CFLAGS=-Wno-int-conversion
+
+common --registry=https://bcr.bazel.build
+
+# =================================
+# Sanitizer
+# =================================
+build:sanitizer --copt -fsanitize=address,undefined,vptr,function
+build:sanitizer --linkopt -fsanitize=address,undefined,vptr,function
+build:sanitizer --linkopt --rtlib=compiler-rt
+build:sanitizer --linkopt -fsanitize-link-c++-runtime
+build:sanitizer --copt -O1
+# seastar has to be run with system allocator when using sanitizers
+build:sanitizer --@seastar//:system_allocator=True
+
+# =================================
+# Security
+# =================================
+build:cfi --copt -fsanitize=cfi
+build:cfi --copt -fvisibility=hidden
+build:cfi --copt -flto
+
+# fortify source requires a high optimization level. when using this feature add
+# `--copt -O2` or combine with a compilation mode like `-c opt`. the default
+# bazel toolchain adds _FORTIFY_SOURCE=1 in `-c opt`, so first clear the
+# definition to avoid -Wmacro-redefined warnings.
+build:fortify-source --copt -U_FORTIFY_SOURCE
+build:fortify-source --copt -D_FORTIFY_SOURCE=2
+
+build:safe-stack --copt -fsanitize=safe-stack
+build:stack-clash --copt -fstack-clash-protection
+build:stack-protector --copt -fstack-protector-strong
+build:relro --copt -fno-plt
+build:relro --linkopt -Wl,-z,relro,-z,now
+
+# Use `--config=secure` for all supported security settings
+build:secure --config=fortify-source
+build:secure --config=stack-clash
+build:secure --config=stack-protector
+build:secure --config=relro
+# Cannot enable until all external deps support
+#build:secure --config=cfi
+#build:secure --config=safe-stack
+
+# =================================
+# Release
+# =================================
+build:release --compilation_mode opt
+build:release --config=secure
+build:release --copt -mllvm --copt -inline-threshold=2500
+build:release --linkopt=-flto
```

Comment:
I don't think this gets enabled in the current release build (and also I don't think it would do anything)

Reply:
thin-lto? i added it by request of @rockwotj, i'll let him chime in. good point about it not being used in the current build!

Reply:
My intent was to preserve this behavior: https://github.com/redpanda-data/redpanda/blob/93474b7b5d51c183f584d888ea2ac618be4c056a/src/v/redpanda/CMakeLists.txt#L90

Does that line to something different or am I misunderstanding what that does?

Reply:
My understanding is that quoted like from cmake applies to only the final binary linking, and doesn't take affect transitively across the entire tree, but would need to be enabled globally to have an affect. i'm also not sure i know what i'm talking about :P

Reply:
I was referring to the vtools build yeah. It needs to be enabled for the compile steps as well yeah. I think passing it only at link time does have some effect like it being able to remove unused functions from the final binary but it's not the full thing that people usually expect.

Hadn't seen that bit in the public build. Are we sure that actually evaluates to true? Just had a quick look and it doesn't get passed for the compile steps at least (not sure whether that is what that cmake logic would do or not).